### PR TITLE
DEPR: provide deprecations and exposure for NaTType

### PR DESCRIPTION
**Pull request by jreback**
xref #16137

### CODE REVIEW

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Currently, under the main module privacy changes table we have

> - The function union_categoricals() is now importable from pandas.api.types, formerly from pandas.types.concat (GH15998)

Do you want to add an entry for `NaTTYpe` is not importable from `pandas.api.types`?

**Comment by jreback** [comment]
> Do you want to add an entry for NaTTYpe is not importable from pandas.api.types?
sure

**Comment by jorisvandenbossche** [comment]
@jreback Have you seen the comment of @shoyer on the issue?

> The right way to use this going forward is probably type(pd.NaT). I'm not sure what the point of exposing NaTType is, given that it's a singleton. Python 2 had types.NoneType but they removed it in Python 3.

I think that makes sense, and thus it is maybe not needed to expose this publicly (we can also say to use that in the deprecation message)

**Comment by jreback** [comment]
if u look at the PR that's exactly what i did

**Comment by jreback** [comment]
it's in the deprecation message

File: pandas/__init__.py [hunk 1]
```diff
@@ -77,6 +77,7 @@
                           moved={'Timestamp': 'pandas.Timestamp',
                                  'Timedelta': 'pandas.Timedelta',
                                  'NaT': 'pandas.NaT',
+                                 'NaTType': 'type(pandas.NaT)',
```
Comment (jreback) [inline pandas/__init__.py]: @jorisvandenbossche

**Comment by jorisvandenbossche** [comment]
Whoops :-) 
That's what you get from rebasing instead of adding a new commit, as there was no notification of a change, I supposed it was the original state that was merged :-)

**Pull request merged**